packages/web-components/tests/e2e-storybook/cypress/integration/carousel/carousel.e2e.js

```diff
@@ -0,0 +1,343 @@
+          );
+        });
+
+      cy.takeSnapshots();
+    });
+  },
+  checkVideoDurationText: () => {
+    it('should render the video duration in the footer', () => {
+      cy.get(_selectors.videoFooter)
+        .find('span.bx--card__cta__copy')
+        .then($duration => {
+          expect($duration).not.to.be.empty;
+        });
+    });
+  },
+  checkSameHeight: () => {
+    it('should have headings all the same height', () => {
+      cy.get(_selectors.heading)
+        .wait(1000)
+        .then($heading => {
+          let headingHeightNum = $heading.first().height();
+          let headingHeight = headingHeightNum + 'px';
+
+          expect($heading).to.not.have.css('height', '0px');
+          expect($heading).to.have.css('height', headingHeight);
+        });
+    });
+  },
+  checkClickableCard: () => {
+    it("should check that the footer's pseudo class takes up entire card to be clickable", () => {
+      cy.get(_selectorBase).then($carousel => {
+        if ($carousel.find(_selectors.footer).length > 0) {
+          cy.get(_selectors.footer)
+            .shadow()
+            .find('a.bx--card__footer')
+            .then($link => {
+              const url = $link.prop('href');
+              expect(url).not.to.be.empty;
+
+              const win = $link[0].ownerDocument.defaultView;
+              const after = win.getComputedStyle($link[0], ':after');
+              const positionValue = after.getPropertyValue('position');
+              const insetValue = after.getPropertyValue('inset');
+
+              expect(positionValue).to.eq('absolute');
+              if (Cypress.browser.name !== 'firefox') {
+                expect(insetValue).to.eq('0px');
+              }
+            });
+        }
+
+        if ($carousel.find(_selectors.videoFooter).length > 0) {
+          cy.get(_selectors.videoFooter)
+            .shadow()
+            .find('a.bx--card__footer')
+            .then($link => {
+              const url = $link.prop('href');
+              expect(url).not.to.be.empty;
+
+              const win = $link[0].ownerDocument.defaultView;
+              const after = win.getComputedStyle($link[0], ':after');
+              const positionValue = after.getPropertyValue('position');
+              const insetValue = after.getPropertyValue('inset');
+
+              expect(positionValue).to.eq('absolute');
+              if (Cypress.browser.name !== 'firefox') {
+                expect(insetValue).to.eq('0px');
+              }
+            });
+        }
+      });
+    });
+  },
+  checkScroll: () => {
+    it('should scroll forward when Next button is clicked and back when the Previous button is clicked', () => {
+      cy.get(_selectors.buttonNext).click();
+
+      cy.takeSnapshots();
+
+      cy.get(_selectors.buttonPrevious).click();
+
+      cy.takeSnapshots();
```

Comment:
Nit: I think you ought to be able to chain the takeSnapshots into these and only need to call on `cy` once.

Not Nit: Does the scroll happen instantly or is it using smoothScroll behavior? If the scroll takes time, we might want to throw a `.wait(1000)` in here to make sure the page has time to scroll before we take snapshots

Reply:
Good points. I chained them and added a wait.